daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilParseOutputStreamContentHandler.scala

```diff
@@ -107,6 +208,36 @@ class DaffodilParseOutputStreamContentHandler(out: OutputStream, pretty: Boolean
     outputNewlineStack.push(false)
   }
 
+  private def outputTagName(uri: String, localName: String, qName: String, atts: Maybe[Attributes] = Nope): Unit = {
+    val tagName = {
+      if (qName.nonEmpty) {
+        qName
+      } else {
+        // use attributes
```

Comment:
This could be a bit more verbose to be more clear what "use attributes" means. I think what's going on here is we don't have a qname string, so we need to figure out what prefix to use so one case uses attributes and the other uses globalPrefixMapping?